public/app/features/explore/Graph/ExploreGraph.tsx

```diff
@@ -166,7 +153,13 @@ export function ExploreGraph({
           hiddenFrames.push(getFrameDisplayName(frame));
         }
       });
-      onHiddenSeriesChangedRef.current(hiddenFrames);
+      if (
+        previousHiddenFrames.current === undefined ||
+        !isEqual(sortBy(hiddenFrames), sortBy(previousHiddenFrames.current))
```

Comment:
Is there any possibility for a performance impact of this conditional with sorting and comparison?

Reply:
It's a one-off operation when data / config changes. Non-scientific benchmark on my laptop shows that it may increase from 0.1ms to 0.2ms. I didn't see any issues while a Loki query response was being split. This list contains names of hidden frames so by default it's empty and it may add some impact only with filtering. Lodash it quite fast at sorting strings so it should be quick even with thousands of data frames.

Reply:
I'll merge it after syncing with main but let us know if you spot anything suspicious 😉

Reply:
For sure. Thank you!